sky/backends/cloud_vm_ray_backend.py

```diff
@@ -784,7 +784,16 @@ def _update_blocklist_on_gcp_error(
         else:
             # No such structured error response found.
             assert not exception_list, stderr
-            if 'was not found' in stderr:
+            if 'Head node fetch timed out' in stderr:
+                # Example: click.exceptions.ClickException: Head node fetch
+                # timed out. Failed to create head node.
+                # This is a transient error, but we have retried in need_ray_up
```

Comment:
nit/Q: is this a transient error from the cloud, or due to certain image sizes being too large (which then makes the timeout in ray flaky)?

Reply:
It is likely the issue is with the cloud, as ray will get through the waiting loop, once the VM goes into `PROVISIONING` state and has the tag set. However, I have seen multiple times when I `sky launch --gpus A100:8`, our program get the IP address for the VM, while the VM just got deleted by GCP on the console.

Reply:
SG